ttnn/cpp/ttnn/operations/data_movement/pad/pad.hpp

```diff
@@ -5,361 +5,41 @@
 #pragma once
 
 #include "ttnn/tensor/types.hpp"
-#include "ttnn/operations/core/core.hpp"
-
-#include "ttnn/run_operation.hpp"
-
-#include "device/pad_op.hpp"
 
 #include <ranges>
 
+#include "ttnn/decorators.hpp"
```

Comment:
but then this header pulls in some heavy things with it

Reply:
Can this not go down into the cpp?   Not seeing why not.

Reply:
@eyonland , I wish!

but we define in a header
```
constexpr auto permute = ttnn::register_operation<"ttnn::permute", ttnn::operations::data_movement::ExecutePermute>();
```